content/blog/2019-10-23-register-allocation/index.md

```diff
@@ -0,0 +1,142 @@
++++
+title = "Register Allocation for Bril"
+extra.author = "Hongbo Zhang, Sachille Atapattu, Wen-Ding Li"
+extra.bio = """
+  Hongbo Zhang is a first PhD student in computer science. He is interested in systems and computer architectures. He is also okay archer shooting recurve bow.
+  [Wen-Ding Li](https://www.cs.cornell.edu/~wdli/) is a Ph.D. student at Cornell interested in high-performance code generation.
+"""
++++
+# Register Allocation for Bril
+
+Processors usually have registers which can be used to store variables for faster access. Program will run faster if we can put variables in these registers. Nevertheless, registers are limited, so often times we still need to access memory (or cache) instead of registers and memory access are costly compared with register access. Although modern program often use large number of variables, we can analyze their live range so the same registers may be able to hold different varialbes throughout the program. Then, we can put more varialbes in the registers which may lead to fewer memory access and get a faster program. 
```

Comment:
Program will run faster -> programs
often times -> oftentimes
memory access are costly -> accesses
modern program often use -> programs
their live range -> ranges
varialbes -> variables (twice)
fewer memory access -> accesses
get a faster program -> *yield* a faster program

Throughout your post, please do these things before requesting another review:
- Run spell check.
- Be careful about plural vs. singular nouns and pronouns.